src/timeMachine/components/dynamicFluxFunctionsToolbar/FluxFunctionsToolbar.tsx

```diff
@@ -8,56 +9,105 @@ import {DapperScrollbars} from '@influxdata/clockface'
+import {getFluxPackages} from 'src/timeMachine/actions/scriptEditorThunks'
 
 // Types
-import {FluxToolbarFunction} from 'src/types'
+import {RemoteDataState} from 'src/types'
+import {AppState} from 'src/types'
 
 interface OwnProps {
-  onInsertFluxFunction: (func: FluxToolbarFunction) => void
+  onInsertFluxFunction: (func) => void
+}
+
+interface DispatchProps {
+  getFluxPackages: () => void
 }
 
-const DynamicFluxFunctionsToolbar: FC<OwnProps> = (props: OwnProps) => {
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & OwnProps & DispatchProps
+
+const DynamicFluxFunctionsToolbar: FC<Props> = (props: Props) => {
   const [searchTerm, setSearchTerm] = useState('')
+  const [fluxServiceError, setFluxServiceError] = useState<RemoteDataState>(
+    RemoteDataState.NotStarted
+  )
 
   const handleSearch = (searchTerm: string): void => {
     setSearchTerm(searchTerm)
   }
 
+  useEffect(() => {
+    const getFluxFuncs = async () => {
+      try {
+        setFluxServiceError(RemoteDataState.Loading)
+
+        if (props.fluxFunctions.length === 0) {
+          await props.getFluxPackages()
+          setFluxServiceError(RemoteDataState.Done)
+        }
+        setFluxServiceError(RemoteDataState.Done)
+      } catch (err) {
+        console.error(err)
+        setFluxServiceError(RemoteDataState.Error)
+      }
+    }
+    getFluxFuncs()
+  }, [])
+
   const handleClickFunction = useCallback(
-    (func: FluxToolbarFunction) => {
+    (func: Fluxdocs) => {
       props.onInsertFluxFunction(func)
     },
     [props.onInsertFluxFunction]
   )
 
   return useMemo(() => {
     return (
-      <ErrorBoundary>
-        <FluxToolbarSearch onSearch={handleSearch} resourceName="Functions" />
-        <DapperScrollbars className="flux-toolbar--scroll-area">
-          <div className="flux-toolbar--list" data-testid="flux-toolbar--list">
-            <TransformToolbarFunctions
-              funcs={FLUX_FUNCTIONS}
-              searchTerm={searchTerm}
+      <SpinnerContainer
+        loading={fluxServiceError}
+        spinnerComponent={<TechnoSpinner />}
+      >
+        <ErrorBoundary>
+          <FluxToolbarSearch onSearch={handleSearch} resourceName="Functions" />
+          <DapperScrollbars className="flux-toolbar--scroll-area">
+            <div
+              className="flux-toolbar--list"
+              data-testid="flux-toolbar--list"
             >
-              {sortedFunctions =>
-                sortedFunctions.map(func => (
-                  <ToolbarFunction
-                    onClickFunction={handleClickFunction}
-                    key={`${func.name}_${func.desc}`}
-                    func={func}
-                    testID={func.name}
-                  />
-                ))
-              }
-            </TransformToolbarFunctions>
-          </div>
-        </DapperScrollbars>
-      </ErrorBoundary>
+              <TransformToolbarFunctions
+                funcs={props.fluxFunctions}
```

Comment:
See comment above regarding destructuring out of the `props` near the top. It is better to destructure this out into its own `fluxFunctions` variable.